lib/rust/ensogl/core/src/display/scene/layer.rs

```diff
@@ -886,6 +908,37 @@ impl SublayersModel {
 
 
 
+// ===================
+// === MaskedLayer ===
+// ===================
+
+/// A layer with attached mask. Every object in `mask` layer will mask the objects of the
+/// `masked_object` layer.
+///
+/// One of the use cases might be an `ensogl_scroll_area::ScrollArea` component
+/// implementation. To clip the area's content (so that it is displayed only inside its borders) we
+/// place the area's content in the `masked_object` layer; and we place a rectangular mask in the
+/// `mask` layer.
+#[derive(Debug, Clone, CloneRef, Deref)]
+#[allow(missing_docs)]
+pub struct MaskedLayer {
+    #[deref]
+    pub masked_object: Layer,
+    pub mask:          Layer,
+}
+
+impl MaskedLayer {
+    /// Constructor. The passed [`camera`] is used to render created layers.
+    pub fn new(logger: &Logger, camera: &Camera2d) -> Self {
+        let masked_object = Layer::new_with_cam(logger.sub("MaskedLayer"), camera);
+        let mask = Layer::new_with_cam(logger.sub("MaskLayer"), camera);
+        masked_object.set_mask(&mask);
+        Self { masked_object, mask }
+    }
+}
```

Comment:
1. I still don't understand why this structure exists. It just creates 2 layers and sets one as a mask for the other one. Maybe this struct has a sense, but I still don't feel it. Why it's better to use this struct instead of just creating a layer, and another one and setting them to mask each other with `layer1.set_mask(&layer2);`?
2. Why a masked layer is called a "masked object"?

Reply:
1. TBH I made a mistake and totally forgot why I left it while refactoring. I was thinking about different scenarios when we want to add new shapes to the `mask_layer` (for example, to enable/disable masking on demand) - it's much easier to work with always available `mask_layer` than without it. Though I was not convinced by these examples and decided to remove the struct. And *then* I released the actual reason why I left it. **`set_mask` does not store a `Layer` smart-pointer internally, it uses `WeakLayer` under the hood!** So we need to store `mask_layer` somewhere or otherwise it would be deallocated. So I added this explanation to the docs.
2. I'm pretty sure you named it during our interactive review call, though I might be wrong. Renamed it to `masked_layer`.